client/src/components/edit-formula-dialog/EditFormulaDialog.tsx

```diff
@@ -5,16 +5,17 @@
 import { AttainmentData, Formula, FormulaData } from 'aalto-grades-common/types';
 import {
   Box, Button, Dialog, DialogTitle, DialogContent, Step, StepLabel, Stepper,
-  Typography
 } from '@mui/material';
 import PropTypes from 'prop-types';
 import { useState, JSX } from 'react';
 
+import FormulaSummary from './FormulaSummary';
 import SelectFormula from './SelectFormula';
 import SetFormulaParams from './SetFormulaParams';
 
 import { useEditAttainment, UseEditAttainmentResult } from '../../hooks/useApi';
 import { State } from '../../types';
+import { sleep } from '../../utils';
 
 export default function EditFormulaDialog(props: {
```

Comment:
What if there are unsaved changes? Should we add the warning so user does not accidentally wipe out unsaved changes if clicking cancel.